src/PrestaShopBundle/Controller/Api/TranslationController.php

```diff
@@ -137,33 +176,26 @@ public function listTreeAction(Request $request)
 
             $search = $request->query->get('search');
 
-            if (in_array($type, ['modules', 'themes']) && empty($selected)) {
-                throw new Exception('This \'selected\' param is not valid.');
+            if (!in_array($type, self::ACCEPTED_TYPES)) {
+                throw new Exception(sprintf("The 'type' parameter '%s' is not valid", $type));
             }
 
-            switch ($type) {
-                case 'themes':
-                    $tree = $this->getNormalTree($lang, $type, $selected, $search);
-                    break;
-
-                case 'modules':
-                    $tree = $this->getModulesTree($lang, $selected, $search);
-                    break;
-
-                case 'mails':
-                    $tree = $this->getMailsSubjectTree($lang, $search);
-                    break;
+            if (in_array($type, [self::TYPE_MODULES, self::TYPE_THEMES]) && empty($selected)) {
+                throw new Exception("The 'selected' parameter is empty.");
+            }
 
-                case 'mails_body':
-                    $tree = $this->getMailsBodyTree($lang, $search);
-                    break;
+            $selectedTheme = (self::TYPE_THEMES === $type) ? $selected : null;
+            $selectedModule = (self::TYPE_MODULES === $type) ? $selected : null;
 
-                default:
-                    $tree = $this->getNormalTree($lang, $type, null, $search);
-                    break;
+            $searchedExpressions = [];
+            if (!is_array($search) && !empty($search)) {
+                $searchedExpressions[] = $search;
             }
 
-            return $this->jsonResponse($tree, $request);
+            return $this->jsonResponse(
+                $this->getTree($lang, $type, $searchedExpressions, $selectedTheme, $selectedModule),
```

Comment:
we have the type, is it mandatory to use `$selectedTheme` and `$selectedModule` since both can't have a value :thinking: (regarding lines 187-188)

Reply:
I could use a variable like $selectedValue but further in the code (like in getTree method), we have to guess what value is in it (module name or theme name), depending on the type. I think it's less clear
I can do the update if you're not agree ;)